packages/dart/npt_flutter/lib/features/tray_manager/widgets/tray_manager.dart

```diff
@@ -22,15 +26,30 @@ class TrayManager extends StatefulWidget {
 class _TrayManagerState extends State<TrayManager> with TrayListener, WindowListener {
   /// Must strongly type [context] here or Dart will infer the wrong type for
   /// the [.read()] extension which causes an error
-  void reloadTray(BuildContext context, _) {
-    context.read<TrayCubit>().reload();
+  void reloadTray(BuildContext context, Loggable state) async {
+    var cubit = context.read<TrayCubit>();
+    switch (state) {
+      case FavoritesState _:
+        cubit.reload(favoriteState: state);
+      case ProfileListState _:
+        cubit.reload(profileListState: state);
+      case ProfilesRunningState _:
+        cubit.reload(profilesRunningState: state);
+      case SettingsLoadedState _:
+        var localizations = await AppLocalizations.delegate.load(state.settings.language.locale);
+        cubit.reload(localizations: localizations);
+      case ProfileState _:
+        cubit.reload(profileState: state);
+      default:
+        cubit.reload(localizations: AppLocalizations.of(context));
+    }
   }
 
   @override
   Widget build(BuildContext context) {
     var trayCubit = context.read<TrayCubit>();
     if (trayCubit.state is TrayInitial) {
-      trayCubit.initialize();
+      trayCubit.initialize(localizations: AppLocalizations.of(context));
```

Comment:
Tray cubit also takes localizations as input in the initialize function, so that the it can immediately set the tray with the desired language on startup.